### PR TITLE
docs: added doc for agoric/synpress commands

### PR DESCRIPTION
**Pull request by frazarshad**
The PR documents the `@agoric/synpress` commands, detailing each command's functionality, the parameters it accepts, and the return values.

To see the markdown output, visit the `/e2e-testing.html#agoric-synpress-commands` route.

### CODE REVIEW

File: main/e2e-testing.md [hunk 1]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+The command takes a single object as an argument with the following keys:
+
+- `secretWords` _(**string**)_: The secret words for the wallet. (default is a predefined mnemonic phrase that represents a wallet address on the Agoric chain)
+- `privateKey` _(**string**)_: The private key for the wallet.
+- `password` _(**string**)_: The password for the wallet. (default is `Test1234`).
+- `newAccount` _(**boolean**)_: Whether to create a new wallet or not. (default is `false`).
+- `walletName` _(**string**)_: The name of the wallet. (default is `My Wallet`).
+- `selectedChains` _(**array**)_: The chains to select. (default is an empty array, `[]`).
+- `createNewWallet` _(**boolean**)_: Whether to create a new wallet (default is `false`).
+
```
Comment (rabi-siddique) [inline main/e2e-testing.md:168]: `newAccount` and `createNewWallet` seem redundant and might cause ambiguity. I've created an issue [over here](https://github.com/agoric-labs/synpress/issues/47) and we will address it soon.

File: main/e2e-testing.md [hunk 2]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
 
 - [PSM dApp E2E Tests](https://github.com/Agoric/dapp-psm/tree/main/tests/e2e)
 - [Wallet dApp E2E Tests](https://github.com/frazarshad/wallet-app/tree/main/test/e2e)
+
+## @agoric/synpress Commands
+
+The `@agoric/synpress` commands are custom Cypress commands designed for use with Keplr. They provide ready-made functions to interact with Keplr-specific elements and workflows.
+
+## `setupWallet`
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:151]: use `###` here and in the following subheadings unless you are using another format.

File: main/e2e-testing.md [hunk 3]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+- `selectedChains` _(**array**)_: The chains to select. (default is an empty array, `[]`).
+- `createNewWallet` _(**boolean**)_: Whether to create a new wallet (default is `false`).
+
+**Returns:**
+
+- _(**boolean**)_: `true` if the wallet was set up successfully, `false` otherwise.
+
+## `acceptAccess`
+
+Handles the authorization process for connecting the Keplr wallet to a DApp. Typically, DApps provide a `Connect Wallet` option to initiate this process.
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:175]: remove Typically.

File: main/e2e-testing.md [hunk 4]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+
+Handles signing transactions within a DApp using the Keplr wallet. This occurs when a user initiates a transaction, such as sending tokens or interacting with a smart contract.
+
+**Returns:**
+
+- _(**boolean**)_: `true` if transaction was approved successfully, `false` otherwise.
+
+## Managing Browser Tabs
+
+In DApp testing, we typically have two important tabs open in the browser: one for the Keplr extension window and another for the Cypress window, where the DApp is running.
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:191]: remove Typically.

File: main/e2e-testing.md [hunk 5]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+
+Switches the focus to the Cypress window tab.
+
+**Returns:**
+
+- _(**boolean**)_:`true` if the focus change was successful, `false` otherwise.
+
+## `disconnectWalletFromDapp`
+
+Disconnects the Keplr wallet from all previously connected DApps. After executing this command, the DApps will need to request permission to connect to the wallet again.
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:232]: remove "the DApps will need to request permission to connect to the wallet again."

File: main/e2e-testing.md [hunk 6]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+
+**Returns:**
+
+- _(**boolean**)_:`true` if the disconnection was successful, `false` otherwise.
+
+## `getWalletAddress`
+
+Retrieves the wallet address associated with a specified blockchain.
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:240]: add "from the currently active wallet."

File: main/e2e-testing.md [hunk 7]
```diff
@@ -143,3 +143,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+
+- `tokenName` _(**string**)_: The name of the token for which you want to retrieve the balance.
+
+**Returns:**
+
+- _(**number**)_: The amount of the specified token currently held in the wallet.
+
+<br />
+
+For a full list of all the commands, you can refer to the [Keplr plugin file](https://github.com/agoric-labs/synpress/blob/dev/plugins/keplr-plugin.js).
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:284]: remove "all the"

File: main/e2e-testing.md [hunk 8]
```diff
@@ -143,3 +141,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
...
+
+### `confirmTransaction`
+
+Handles signing transactions within a DApp using the Keplr wallet. This occurs when a user initiates a transaction, such as sending tokens or interacting with a smart contract.
+
+**Returns:**
+
+- _(**boolean**)_: `true` if transaction was approved successfully, `false` otherwise.
+
+### Managing Browser Tabs
```
Comment (mudassir-agoric) [inline main/e2e-testing.md:187]: For consistency make this ## and its subheading commands ### or in the previous section do the opposite: make the heading ### and commands ####.

**Review by mudassir-agoric — CHANGES_REQUESTED** [review]

File: main/e2e-testing.md [hunk 9]
```diff
@@ -143,3 +141,142 @@ Some examples projects utilizing `@agoric/synpress` for e2e tests:
 
 - [PSM dApp E2E Tests](https://github.com/Agoric/dapp-psm/tree/main/tests/e2e)
 - [Wallet dApp E2E Tests](https://github.com/frazarshad/wallet-app/tree/main/test/e2e)
+
+## @agoric/synpress Commands
+
+The `@agoric/synpress` commands are custom Cypress commands designed for use with Keplr. They provide ready-made functions to interact with Keplr-specific elements and workflows.
```
Comment (ansabgillani) [inline main/e2e-testing.md:147]:
```suggestion
The `@agoric/synpress` is a custom command set for Cypress designed for use with Keplr. They provide ready-made functions to interact with Keplr-specific elements and workflows.
```

Comment (rabi-siddique) [reply main/e2e-testing.md:147]: Wouldn't `@agoric/synpress  is a custom command`  imply that it's just custom commands for interacting with Keplr, when in fact it's a full testing framework?

**Review by mudassir-agoric — APPROVED** [review]

**Pull request merged**